libs/cmdio/exec.go

```diff
@@ -0,0 +1,162 @@
+package cmdio
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+)
+
+type Executor struct {
+	dir         string
+	scriptFiles []string
+}
+
+func NewCommandExecutor(dir string) *Executor {
+	return &Executor{
+		dir:         dir,
+		scriptFiles: nil,
+	}
+}
+
+func (e *Executor) StartCommand(ctx context.Context, command string) (*exec.Cmd, io.Reader, error) {
+	interpreter, err := wrapInShellCall(command)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return e.start(ctx, interpreter)
+}
+
+func (e *Executor) start(ctx context.Context, i *interpreter) (*exec.Cmd, io.Reader, error) {
+	e.scriptFiles = append(e.scriptFiles, i.scriptFile)
+	cmd := exec.CommandContext(ctx, i.executable, i.args...)
+	cmd.Dir = e.dir
+
+	outPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	errPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return cmd, io.MultiReader(outPipe, errPipe), cmd.Start()
```

Comment:
If we can wait for process completion from a goroutine then we can remove the `scriptFile` directly and don't need to have the additional cleanup calls.

Reply:
Regarding stdout and stderr -- there will be valid cases for consuming them separately. How about we return both separately and let callers apply the `io.MultiReader` if they want to?